public/pages/workflows/workflow_list/resource_list.tsx

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect, useCallback } from 'react';
+import { columns } from '../../workflow_detail/tools/resources/columns';
+import {
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiBasicTable,
+  EuiButtonIcon,
+  RIGHT_ALIGNMENT,
+  EuiText,
+  Direction,
+  EuiEmptyPrompt,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import {
+  Workflow,
+  WorkflowResource,
+  customStringify,
+} from '../../../../common';
+import { AppState, useAppDispatch } from '../../../store';
+import { fetchResourceData, getDataSourceId } from '../../../utils';
+import { useSelector } from 'react-redux';
+import { isEmpty } from 'lodash';
+
+interface ResourceListProps {
+  workflow?: Workflow;
+}
+
+/**
+ * The searchable list of resources for a particular workflow.
+ */
+export function ResourceList(props: ResourceListProps) {
+  const [allResources, setAllResources] = useState<WorkflowResource[]>([]);
+  const dispatch = useAppDispatch();
+  const dataSourceId = getDataSourceId();
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<{
+    [key: string]: any;
+  }>({});
+  const loading = useSelector((state: AppState) => state.opensearch.loading);
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+  const [sortField, setSortField] = useState<keyof WorkflowResource>('id');
+  const [sortDirection, setSortDirection] = useState<Direction>('asc');
+
+  // Hook to initialize all resources. Reduce to unique IDs, since
+  // the backend resources may include the same resource multiple times
+  // (e.g., register and deploy steps persist the same model ID resource)
+  useEffect(() => {
+    if (props.workflow?.resourcesCreated) {
+      const resourcesMap: { [id: string]: WorkflowResource } = {};
+      props.workflow.resourcesCreated.forEach((resource) => {
+        resourcesMap[resource.id] = resource;
+      });
+      setAllResources(Object.values(resourcesMap));
+    }
+  }, [props.workflow?.resourcesCreated]);
+
+  // Renders the expanded row to show resource details in a code block.
+  const renderExpandedRow = useCallback(
+    (data: any) => (
+      <EuiFlexGroup direction="column" gutterSize="xs">
+        <EuiFlexItem grow={true} style={{ paddingLeft: '20px' }}>
+          <EuiText size="m">
+            <h4>Resource details</h4>
+          </EuiText>
+        </EuiFlexItem>
+        <EuiFlexItem grow={true} style={{ paddingLeft: '20px' }}>
+          {!data.startsWith('error:') && !loading ? (
```

Comment:
This is not a clean way to propagate errors. Additionally, the input `data` is not typesafe / defined as a string, so we cannot assume we can leverage `startsWith()`. 

This is another situation where this state can be much more cleanly passed via the redux store, the same way as the `loading` var is. In fact, you already have the `errorMessage` var populated. Let's leverage this, and then just dynamically display the `EuiEmptyPrompt` with the error message if `errorMessage` is defined.